### PR TITLE
Replace Mask

### PR DESCRIPTION
**Pull request by mckadesorensen**
Added functionality for replacing mask with networks prediction in the test_wrapper.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -51,8 +51,20 @@ def test_wrapper(args: Namespace) -> None:
         print("ERROR: This dataset is not compatible with your model")
         return
     if dataset_type(args.dataset) == ModelType.MASKED:
-        predictions, test_iter = test_model_masked(model, args.dataset)
-        plot_masked_predictions(predictions, test_iter, args.dataset)
+        if args.replace is True:
```
Comment (Askaholic) [inline main.py]:
```suggestion
        if args.replace:
```

File: src/asf_cnn.py [hunk 2]
```diff
@@ -130,8 +131,12 @@ def test_model_binary(model: Model, dataset: str, verbose: int = 1
     return details, confusion_matrix
 
 
-def test_model_masked(model: Model, dataset: str,
-                      verbose: int = 1) -> Tuple[np.ndarray, Iterator]:
+def test_model_masked(
+    model: Model,
+    dataset: str,
+    t_f: bool,
```
Comment (Askaholic) [inline src/asf_cnn.py]: what is `t_f`?

File: src/asf_cnn.py [hunk 3]
```diff
@@ -140,11 +145,23 @@ def test_model_masked(model: Model, dataset: str,
     if verbose > 0:
         model.summary()
 
-    _, test_iter = load_dataset_masked(dataset)
-    predictions = model.predict_generator(
-        test_iter, len(test_iter), verbose=verbose
-    )
-    test_iter.reset()
-    masked_predictions = predictions.round(decimals=0, out=None)
+    if t_f is True:
```
Comment (Askaholic) [inline src/asf_cnn.py]:
```suggestion
    if t_f:
```

File: src/asf_cnn.py [hunk 4]
```diff
@@ -140,11 +145,23 @@ def test_model_masked(model: Model, dataset: str,
...
+        return masked_predictions, dataset_data[0], dataset_data[1]
+
+    else:
+        _, test_iter = load_dataset_masked(dataset)
+        predictions = model.predict_generator(
+            test_iter, len(test_iter), verbose=verbose
+        )
+        test_iter.reset()
+        masked_predictions = predictions.round(decimals=0, out=None)
+        print(type(masked_predictions))
```
Comment (Askaholic) [inline src/asf_cnn.py]: Delete?

File: src/dataset/masked.py [hunk 5]
```diff
@@ -46,8 +46,28 @@ def load_dataset(dataset: str) -> Tuple[Iterator, Iterator]:
     return train_iter, test_iter
 
 
-def make_metadata(dataset: str
-                  ) -> Tuple[MaskedDatasetMetadata, MaskedDatasetMetadata]:
+def load_replace_data(dataset: str) -> Tuple[Iterator, MaskedDatasetMetadata]:
+    replace_gen = ImageDataGenerator(rescale=10)
```
Comment (Askaholic) [inline src/dataset/masked.py]: Removing rescale at some point?

File: src/dataset/masked.py [hunk 6]
```diff
@@ -46,8 +46,28 @@ def load_dataset(dataset: str) -> Tuple[Iterator, Iterator]:
...
+    metadata, _ = make_metadata(dataset, True)
```
Comment (Askaholic) [inline src/dataset/masked.py]: Use keyword argument for boolean arg for readability

File: src/dataset/masked.py [hunk 7]
```diff
@@ -70,10 +90,14 @@ def make_metadata(dataset: str
             )
             folder = os.path.basename(dirpath)
 
-            if folder == 'train':
+            if t_f is True:
```
Comment (Askaholic) [inline src/dataset/masked.py]:
```suggestion
            if t_f:
```

File: src/plots_masked.py [hunk 8]
```diff
@@ -42,6 +48,42 @@ def close_plot(_: Any) -> None:
             nonlocal done
             done = True
 
-        _cbtn = close_button(close_plot)
-        maximize_plot()
-        plt.show()
+        if t_f is True:
```
Comment (Askaholic) [inline src/plots_masked.py]:
```suggestion
        if t_f:
```

File: src/asf_cnn.py (hunk 2 above)
Comment (mckadesorensen) [reply src/asf_cnn.py]: truly_fascinating

File: src/plots_masked.py [hunk 9]
```diff
@@ -2,86 +2,174 @@
...
+               cmap=plt.get_cmap('gist_gray'))
+
+
+def save_img(f_paths: List[str], pred) -> None:
+    with gdal_open(f_paths[2]) as f:
+        mask_projection = f.GetProjection()
+        mask_geo_transform = f.GetGeoTransform()
+
+    write_mask_to_file(
+        pred.reshape(512, 512), f_paths[2], mask_projection, mask_geo_transform
```
Comment (gtmeier) [inline src/plots_masked.py]: maybe change 512 to a variable to keep code DRY

Comment (mckadesorensen) [reply src/plots_masked.py]: I'll change that in the plots function since reshape is used more than once.

File: src/dataset/masked.py (hunk 5 above)
Comment (mckadesorensen) [reply src/dataset/masked.py]: Just for the books, I did, then we decided to keep it.

**Pull request merged**